### PR TITLE
Significantly improve custom set upload UX

### PR DESCRIPTION
**Pull request by cognivore**
## Linked tickets
No linked ticket.

## Explanation of the issue
It's hard for MSE users to playtest their sets on dr4fts. For instance, even if they would host images on some FTP, it would require them a lot of manual work to modify XML by hand.

## Description of your changes
We provide a complementary script for MSE users they can call to validate their XML for compliance with dr4ft, as well as the **a script they can call to inject picurls** making playtesting more fun!

It also adds a check for things that are tokens and short-ciruits cards.card loop in parser.js.

This PR also significantly improves the upload UX by saying which set number in XML broke the upload and which card name didn't pass checks.

## Screenshots

image

### CODE REVIEW

**Comment by cognivore** [comment]
To glimpse then, untangle burnsPerPack from just cube.

[This commit does it](https://github.com/cognivore/dr4ft/commit/d24714242a969efa6245a2dad46b461e4d89853b), but it's very noisy because I edit in VSCode and it formats stuff by default (which is a good thing).

I suggest incorporating linked changes into upstream 🙇

**Comment by cognivore** [comment]
Oh my, Github added the third commit to the pull request. It's annoying. If nobody will cherry-pick it in a while, I'll close it and will re-open with just the first two commits.

**Comment by tooomm** [comment]
Hi @cognivore

>Oh my, Github added the third commit to the pull request. It's annoying. 

Your PR is updated because you added new commits to the branch this PR is created from.
Always keep the master/main branch of your fork in sync with the upstream repository. That also means that you should not add your own commits to that branch and don't create a PR from there.
Just create a new branch for every feature you are working on and group related commits in that branch. That way you can also work in parallel on several features independently and maintain a PR for each of them.

<br>

```
TLDR

In summary, if you want to contribute to a project, the simplest way is to:

 1) Find a project you want to contribute to
 2) Fork it
 3) Clone it to your local system
 4) **Make a new branch**
 5) Make your changes *in that branch*
 6) Push it back to your repo
 7) *Create PR from your new branch*
```
from https://opensource.com/article/19/7/create-pull-request-github

**Comment by tooomm** [comment]
Do you need help with resubmitting your changes? It can be a bit tricky sometimes to deal with git and GitHub. 🙈 

Let me know where you are at, and we can try to help! @cognivore

**Comment by cognivore** [comment]
No, thanks.
I'm from being a stranger to Git, but Github logic took me by surprise.
The fact that I made a PR and then it go changed outside of feature branch
was annoying.

**Comment by tooomm** [comment]
Are you still planning to commit your suggestions? @cognivore 

>The fact that I made a PR and then it go changed outside of feature branch
was annoying.

That explains your issue. You did not use a feature branch in the first place here.
All your changes are placed in the master branch of your fork which is the basis of this PR.
GitHub updates open PR's if their base branch receives new commits.

**Comment by cognivore** [comment]
No plans to maintain this patchset do far, you can close the PR.
